client/src/pages/Room/Room.tsx

```diff
@@ -145,12 +153,15 @@ const Room: React.FC = () => {
     });
   };
 
-  const handleScroll = useCallback((e: any) => {
-    e.persist();
-    if (bodyRef.current.scrollTop === 0) {
-      fetchMoreMessages();
-    }
-  }, []);
+  const handleScroll = useCallback(
+    (e: any) => {
+      e.persist();
+      if (bodyRef.current.scrollTop === 0) {
+        fetchMoreMessages();
+      }
+    },
+    [roomData?.messages?.messages?.length]
```

Comment:
Fixed: not providing dep array would cause error because fetchMoreMessages relays on "roomData?.messages?.messages?.length"